pkg/helpers/newapp/app/app.go

```diff
@@ -167,6 +168,9 @@ func (r *SourceRef) BuildSource() (*buildv1.BuildSource, []buildv1.BuildTriggerP
 			URI: r.URL.StringNoFragment(),
 			Ref: r.URL.URL.Fragment,
 		}
+		if r.SSHPrefixWorkaround {
+			source.Git.URI = "ssh://" + source.Git.URI
```

Comment:
This feels like a band-aid on top of a thing we broke in s2i.

We should fix s2i so that handles regular SSH URIs (which are valid URIs) and SCP style URIs.

Reply:
as I put a hold on in the description, and mentioned wanting to discuss this, I agree this is a band aid ... of course your additional hold does not hurt anything :-)

that said, the reason I want to discuss voice to voice in office hours is 
a) fixing it this in s2i could prove trickier and more time consuming than we realize ... not 100% sure but I want to talk through it
b) if that pans out, is the band aid sufficient given what we have to do elsewhere

Reply:
In particular, the wide use of the embedded golang URL inside the S2I URL throughout new-app / new-build code gave me some pause @adambkaplan 

And that golang URL still can't deal with ssh://

My hope is bring up the code in an editor, walk through the references during office hours, and see what change makes sense where.